### PR TITLE
[ci] increase valgrind timeout to 6 hours

### PR DESCRIPTION
**Pull request by jameslamb**
Blocks #5952.

* upgrades the timeout on the `r-valgrind` CI job to 6 hours, to try to fix https://github.com/microsoft/LightGBM/pull/5952#issuecomment-1613764158

### Notes for Reviewers

As described in #5414, this has to be merged to `master` to be tested, since the valgrind job's configuration is only sourced from `master`.

6 hours is the absolute maximum GitHub Actions allows: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits.

So if this doesn't work, we'll have to come up with another strategy for running the valgrind tests faster (like maybe finding a source of Linux precompiled binaries for `{lightgbm}`'s dependencies), or will have to rely on running them somewhere else (even maybe just locally on a maintainer's laptop 😬 ).

### CODE REVIEW

**Comment by jameslamb** [comment]
`check-docs` is failing with this error that seems to be occuring while rendering the sphinx docs to HTML

> Warning, treated as error:
/home/runner/work/LightGBM/LightGBM/python-package/lightgbm/dask.py:docstring of lightgbm.dask.DaskLGBMClassifier:1:undefined label: 'metadata_routing'

([build link](https://github.com/microsoft/LightGBM/actions/runs/5424541292/jobs/9864047043?pr=5953))

I see that on `master` too ([build link](https://github.com/microsoft/LightGBM/actions/runs/5423459086/jobs/9861417030)).

Since that blocks merging, I'll try to fix it in this PR too. Moving this back to draft until that's fixed.

File: docs/env.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: docs-env
+iname: docs-env
```
Comment (jameslamb) [inline docs/env.yml:1]:
```suggestion
name: docs-env
```

typo

**Comment by jameslamb** [comment]
@guolinke @shiyu1994 could you please review this? So we can continue testing v4.0.0 on #5952.

**Review by jmoralez — APPROVED** [review]

**Comment by jameslamb** [comment]
thanks @jmoralez !

hopefully 6 hours is enough 🤞🏻

**Review by guolinke — APPROVED** [review]

**Pull request merged**